app/scripts/controllers/alert-controller.test.ts

```diff
@@ -28,48 +27,54 @@ const EMPTY_ACCOUNT = {
     importTime: 0,
   },
 };
-describe('AlertController', () => {
-  let controllerMessenger: ControllerMessenger<
-    AlertControllerActions | AllowedActions,
-    | AlertControllerEvents
-    | KeyringControllerStateChangeEvent
-    | SnapControllerStateChangeEvent
-    | AllowedEvents
-  >;
-  let alertController: AlertController;
-
-  beforeEach(() => {
-    controllerMessenger = new ControllerMessenger<
-      AllowedActions,
-      AllowedEvents
-    >();
-    controllerMessenger.registerActionHandler(
-      'AccountsController:getSelectedAccount',
-      () => EMPTY_ACCOUNT,
-    );
 
-    const alertMessenger = controllerMessenger.getRestricted({
+const setupController = ({
```

Comment:
nit: we could use the `withController` pattern we have been using for [other controllers](https://github.com/MetaMask/metamask-extension/blob/fa97321a9c74dce1e6b05e8c07278c956f2cd86f/app/scripts/controllers/account-tracker-controller.test.ts#L88-L88)